apps/meteor/client/views/room/Header/RoomHeaderE2EESetup.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { lazy } from 'react';
+
+import { E2EEState } from '../../../../app/e2e/client/E2EEState';
+import { E2ERoomState } from '../../../../app/e2e/client/E2ERoomState';
+import { useE2EERoomState } from '../hooks/useE2EERoomState';
+import { useE2EEState } from '../hooks/useE2EEState';
+import DirectRoomHeader from './DirectRoomHeader';
+import RoomHeader from './RoomHeader';
+import type { RoomHeaderProps } from './RoomHeader';
+
+const RoomToolboxE2EESetup = lazy(() => import('./RoomToolbox/RoomToolboxE2EESetup'));
+
+const RoomHeaderE2EESetup = ({ room, topic = '', slots = {} }: RoomHeaderProps) => {
+	const e2eeState = useE2EEState();
+	const e2eRoomState = useE2EERoomState(room._id);
+
+	if (e2eeState === E2EEState.SAVE_PASSWORD || e2eeState === E2EEState.ENTER_PASSWORD || e2eRoomState === E2ERoomState.WAITING_KEYS) {
+		return <RoomHeader room={room} topic={topic} slots={slots} roomToolbox={<RoomToolboxE2EESetup />} />;
+	}
+
+	if (room.t === 'd' && (room.uids?.length ?? 0) < 3) {
```

Comment:
Let's use the helper here: https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/core-typings/src/IRoom.ts#L139

Also, what uids being greater than 3 mean? Would it be better to have this in a named variable maybe?